src/main/java/com/endava/internship/service/UserServiceImpl.java

```diff
@@ -2,68 +2,90 @@
 
 import com.endava.internship.domain.Privilege;
 import com.endava.internship.domain.User;
-import com.endava.internship.service.UserService;
-
+import org.apache.commons.lang3.tuple.Pair;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
+import java.util.Map;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class UserServiceImpl implements UserService {
 
     @Override
     public List<String> getFirstNamesReverseSorted(List<User> users) {
-        throw new UnsupportedOperationException("Not implemented");
+        return users.stream().map(u -> u.getFirstName()).sorted((String u1, String u2) -> (u2.compareTo(u1))).collect(Collectors.toList());
```

Comment:
instead 'u -> u.getFirstName()' use please method reference ClassName::methodName
for sorting you ca use Comparator.reverseOrder()

Reply:
resolved